test/test_autograd.py

```diff
@@ -611,6 +611,78 @@ def backward(ctx, grad_b):
 
         TestFn.apply(b).sum().backward()
 
+    def test_free_deep_graph(self):
+        def scope():
+            depth = 150000
+            x = torch.randn(1, requires_grad=True)
+            y = x.clone()
+
+            # build a "chain" computation graph
+            for i in range(depth):
+                y = y + y * 0.000001
+
+            # triggers graph deletion
+            del x
```

Comment:
How does `x` trigger a graph deletion? It's a leaf!

Reply:
Ah thank you for the catch! Followup PR coming soon.